pkg/crypto/crypto_test.go

```diff
@@ -126,6 +128,49 @@ func TestEncryptDecrypt(t *testing.T) {
 	})
 }
 
+func TestUpgradePathEncryptDecrypt(t *testing.T) {
+	simplePassphrase := []byte("secret key")
+
+	testCases := []struct {
+		name           string
+		input          string
+		encryptKey     []byte
+		decryptKey     []byte
+		expectedOut    string
+		expectedEncErr bool
+		expectedDecErr bool
+	}{
+		// error cases
```

Comment:
This is a happy path test case, not error case. Speaking of error cases, I think it would be beneficial to add one. Just a negative test case where you pass in a different key to decrypt and makes sure it returns an error.

Reply:
+1